core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -253,6 +253,38 @@ object UnitAutomation {
         unit.movement.headTowards(encampmentToHeadTowards)
         return true
     }
+    
+    private fun tryRetreat(unit: MapUnit): Boolean {
+        if (!unit.civ.isAtWar()) return false
+        // Precondition: This must be a military unit
+        if (unit.isCivilian()) return false
+        // Better to do a more healing oriented move then
+        if (unit.getDistanceToEnemyUnit(6, true) > 4) return false
+        
+        if (unit.baseUnit.isAirUnit()) {
+            return false
+        }
+
+        val unitDistanceToTiles = unit.movement.getDistanceToTiles()
+        val swapableTiles = unitDistanceToTiles.keys.filter { it.militaryUnit != null && it.militaryUnit!!.owner == unit.owner}.reversed()
+        for (swapTile in swapableTiles) {
+            val otherUnit = swapTile.militaryUnit!!
+            if (otherUnit.health > 80 
+                && unit.getDistanceToEnemyUnit(6, false) < otherUnit.getDistanceToEnemyUnit(6,false)) {
```

Comment:
Unit distance is the same for all swappable tiles, put it outside the for loop

Reply:
How necessary is that since getDistanceToEnemyUnit caches the result?

Reply:
If that's the case, not necessary ☺️